src/jvmMain/kotlin/com/charleskorn/kaml/Yaml.kt

```diff
@@ -1,119 +0,0 @@
-/*
-
-   Copyright 2018-2023 Charles Korn.
-
-   Licensed under the Apache License, Version 2.0 (the "License");
-   you may not use this file except in compliance with the License.
-   You may obtain a copy of the License at
-
-       https://www.apache.org/licenses/LICENSE-2.0
-
-   Unless required by applicable law or agreed to in writing, software
-   distributed under the License is distributed on an "AS IS" BASIS,
-   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-   See the License for the specific language governing permissions and
-   limitations under the License.
-
-*/
-
-package com.charleskorn.kaml
-
-import kotlinx.serialization.DeserializationStrategy
-import kotlinx.serialization.SerializationStrategy
-import kotlinx.serialization.StringFormat
-import kotlinx.serialization.modules.SerializersModule
-import kotlinx.serialization.serializer
-import org.snakeyaml.engine.v2.api.StreamDataWriter
-import org.snakeyaml.engine.v2.api.YamlOutputStreamWriter
-import java.io.IOException
-import java.io.InputStream
-import java.io.InputStreamReader
-import java.io.OutputStream
-import java.io.Reader
-import java.io.StringReader
-import java.io.StringWriter
-import java.nio.charset.Charset
-
-public actual class Yaml actual constructor(
-    override val serializersModule: SerializersModule,
-    public actual val configuration: YamlConfiguration,
-) : StringFormat {
-    override fun <T> decodeFromString(deserializer: DeserializationStrategy<T>, string: String): T {
-        return decodeFromReader(deserializer, StringReader(string))
-    }
-
-    public fun <T> decodeFromStream(deserializer: DeserializationStrategy<T>, source: InputStream, charset: Charset = Charsets.UTF_8): T {
```

Comment:
I removed too much from the JVM source set. If we want to preserve the JVM API, we need to revive some of this code.

Reply:
I'd like to preserve the JVM API, so yes, please keep whatever's required to maintain the API as-is.

Reply:
Sure, working on it!